File: Algoritm by CPP/Algoritm by CPP/Array/10808.cpp
//
//  main.cpp
//  Algoritm by CPP
//
//  Created by zaehorang on 2024/02/12.
//  https://www.acmicpc.net/problem/10808

#include <bits/stdc++.h>
using namespace std;

int freq[26]; // 전역 변수로 생성 시 0으로 채워진다.

int main(void) {
  ios::sync_with_stdio(0);
  cin.tie(0);
    
  string s;
  cin >> s;
    
  for(auto c : s)  // 타입 추론 "auto"
    freq[c-'a']++; // Swift에서는 문자와 문자열간의 산술 연산자 지원 x
    
  for(int i = 0; i < 26; i++)
    cout << freq[i] << ' ';
}

File: Algorithm by Swift/Algorithm by Swift/Array/10808.swift
//
//  main.swift
//  Algorithm by Swift
//
//  Created by zaehorang on 2024/02/12.
//  https://www.acmicpc.net/problem/10808

import Foundation

//MARK: - 참고 풀이
// 알파벳 아스키 코드를 고려하면 for문 한 번으로도 가능.
// O(n)
let input = readLine()!
var array = [Int](repeating: 0, count: 26)
var a:Character = "a"

for alphabet in input {
    let index = Int(alphabet.asciiValue! - a.asciiValue!)  // 대문자가 있을 시 이렇게 사용하면 안된다. (overflow 발생)
    array[index] += 1
}

print(array.map{ String($0) }.joined(separator: " "))


//MARK: - 첫 번째 풀이
// 이중 for문으로 O(n^2)

//var input = readLine()
//
//if let input = input {
//    var answer = [Int](repeating: 0, count: 26)
//    var alphabets = "abcdefghijklmnopqrstuvwxyz"
//    var index = 0
//
//    for str in input {
//
//        for alphabet in alphabets {
//            if str == alphabet {
//                answer[index] += 1
//
//                index = 0
//                break
//            } else {
//                index += 1
//            }
//        }
//    }
//
//    print(answer.map { String($0) }.joined(separator:" ") )
//}
